internal/service/evidently/wait.go

```diff
@@ -0,0 +1,59 @@
+package evidently
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/cloudwatchevidently"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func waitProjectCreated(conn *cloudwatchevidently.CloudWatchEvidently, nameOrARN string, timeout time.Duration) (*cloudwatchevidently.GetProjectOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{},
+		Target:  []string{cloudwatchevidently.ProjectStatusAvailable},
+		Refresh: statusProject(conn, nameOrARN),
+		Timeout: timeout,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*cloudwatchevidently.GetProjectOutput); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+func waitProjectUpdated(conn *cloudwatchevidently.CloudWatchEvidently, nameOrARN string, timeout time.Duration) (*cloudwatchevidently.GetProjectOutput, error) { //nolint:unparam
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{cloudwatchevidently.ProjectStatusUpdating},
+		Target:  []string{cloudwatchevidently.ProjectStatusAvailable},
+		Refresh: statusProject(conn, nameOrARN),
+		Timeout: timeout,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*cloudwatchevidently.GetProjectOutput); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+func waitProjectDeleted(conn *cloudwatchevidently.CloudWatchEvidently, nameOrARN string, timeout time.Duration) (*cloudwatchevidently.GetProjectOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{cloudwatchevidently.ProjectStatusAvailable},
+		Target:  []string{cloudwatchevidently.ErrCodeResourceNotFoundException},
```

Comment:
Since there is not defined enum for a deleted state, we should make the target for deletion an empty slice. The status function will return `""` if the resource is not found.

```suggestion
		Target:  []string{},
```